src/Builder/DriverFactoryInterface.php

```diff
@@ -9,5 +9,5 @@
 
 interface DriverFactoryInterface
 {
-    public function createDriver(array $metadataDirs, Reader $annotationReader): DriverInterface;
+    public function createDriver(array $metadataDirs, ?Reader $annotationReader = null): DriverInterface;
```

Comment:
This one line is why there is a B/C break; widening the annotation reader argument to allow null causes compile errors in classes if they don't do the same.

Reply:
I think it should not be a big issues. JMS Bundle seems to not using those classes, so it will affect only people who is using package without Bundle.